### ISSUE TITLE
Array of objects with different keys will not return all of them even if specified headers

### ISSUE THREAD

**Issue by sagivo** [knrz/CSV.js#40]
```js
new CSV([{a:1, b:2}, {a:1, b:2, c:3}], {
  header: ['a','b','c'],
}).encode();
```
will return:
```
"a","b","c"
1,2
1,2
```
value "c" will not render

same apply if first value is null:
```js
new CSV([{a:1, b:2, c: null}, {a:1, b:2, c:3}], {
  header: ['a','b','c'],
}).encode();
```

**Comment by CyrusNuevoDia** [comment]
That's a good case you found there! 

CSV.js uses the first object as a prototype for the keys, and caches those values. But it should do what you're talking about: Specifying a header should work.

**Comment by bryceg** [comment]
Yeah we keep running into this also when parsing user input data.  Working around it for now by putting in a dummy first row containing all the columns, parsing, then removing the first row.
